client/src/components/QAModule/QuestionsAndAnswers.jsx

```diff
@@ -22,20 +24,73 @@ export default function QuestionsAndAnswers() {
       axios.get(url, { headers })
         .then((res) => {
           const { results } = res.data;
-          console.log(results);
-          setQuestions(results);
+          const sortedResults = results.sort((a, b) => (
+            b.question_helpfulness - a.question_helpfulness
+          ));
+          setQuestions(sortedResults);
+          setFilteredQuestions(sortedResults);
         })
         .catch((err) => {
+          // eslint-disable-next-line no-console
           console.error('Error fetching product data: ', err);
         });
     }
   }, [productId]);
 
+  const handleSearch = (searchTerm) => {
+    if (searchTerm) {
+      const filtered = questions.filter((question) => (
+        question.question_body.toLowerCase().includes(searchTerm.toLowerCase())
+      ));
+      setFilteredQuestions(filtered);
+    } else {
+      setFilteredQuestions(questions);
+    }
+  };
+
+  const openModal = () => {
+    setIsModalOpen(true);
+  };
+
+  const closeModal = () => {
+    setIsModalOpen(false);
+  };
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+  };
+
   return (
     <div>
       <h2>QUESTIONS & ANSWERS</h2>
-      <SearchBar />
-      <QuestionsList questions={questions} />
+      <SearchBar onSearch={handleSearch} />
+      <QuestionsList questions={filteredQuestions} />
+      <div>
+        <button type="button" onClick={openModal}>
+          ADD A QUESTION +
+        </button>
+        <ModalTest isOpen={isModalOpen} onClose={closeModal}>
+          <h2>Ask Your Question</h2>
+          <h4>{`About the [PRODUCT NAME HERE ID: ${productId}]`}</h4>
```

Comment:
Be sure to remove the '[PRODUCT NAME HERE ID ' from this string to only render the product name.